library/include/modules/EventManager.h

```diff
@@ -28,8 +29,12 @@ namespace DFHack {
                 UNIT_NEW_ACTIVE,
                 UNIT_DEATH,
                 ITEM_CREATED,
-                BUILDING,
-                CONSTRUCTION,
+                BUILDING, // todo: deprecate this event
+                BUILDING_CREATED,
+                BUILDING_DESTROYED,
```

Comment:
I suggest putting the destroyed/removed events above the related created/added events. From experience it's important to know what's gone before learning what's new.

Reply:
I just retained the old behaviour of finding new before then finding missing

Reply:
Ok so I just re-ordered them. I didn't remove the deprecation todo's though, because the created/destroyed lists need to be processed before the events are sent; it was done this way to avoid looping the handler's twice and having one where the last tick is updated and the other not...  I mean it's really just a readability thing, so I can change that if you request it.

So bottom line currently if I put the split events above the 2-in-1 events the split ones would technically work better, but as is they're at least consistent with the 2-in-1 versions. Which don't really care which events are sent first since both lists are finalized by the time the events start sending.

I'm going to hold off on pushing until the CI changes are done. I can if you really want me to though.

Reply:
Just to be clear, this was the ordering I was suggesting:
BUILDING_DESTROYED
BUILDING_CREATED
BUILDING
CONSTRUCTION_REMOVED
CONSTRUCTION_ADDED
CONSTRUCTION

This will allow us to avoid bugs like the one currently in the buildings cache - described at #1984 (at the bottom of the description comment). Once the split events are available, I'll migrate the buildings cache to use them, but they will only be useful if destroyed events come before created events.

Reply:
Interesting bug. I'll drop the consistency since it sounds like putting the split events first allows you (and others?) to transition to them better.

Reply:
Gah, sorry, to be even clearer, it's just the relative positions of 
BUILDING_DESTROYED
BUILDING_CREATED
that really matter here. The original BUILDING event can go before or after those two. Or even between. As long as BUILDING_DESTROYED goes before BUILDING_CREATED

Reply:
If `BUILDING` came first then the order of the other two wouldn't matter (literally). Their processing would be redundant so nothing would change in the containers holding the events. I've put the destroyed first for each, then the created, then the old events